sdks/python/apache_beam/runners/direct/executor.py

```diff
@@ -291,44 +294,62 @@ def call(self):
 
     side_input_values = [self._side_input_values[side_input]
                          for side_input in self._applied_ptransform.side_inputs]
-
-    try:
-      evaluator = self._transform_evaluator_registry.get_evaluator(
-          self._applied_ptransform, self._input_bundle,
-          side_input_values, scoped_metrics_container)
-
-      if self._fired_timers:
-        for timer_firing in self._fired_timers:
-          evaluator.process_timer_wrapper(timer_firing)
-
-      if self._input_bundle:
-        for value in self._input_bundle.get_elements_iterable():
-          evaluator.process_element(value)
-
-      with scoped_metrics_container:
-        result = evaluator.finish_bundle()
-        result.logical_metric_updates = metrics_container.get_cumulative()
-
-      if self._evaluation_context.has_cache:
-        for uncommitted_bundle in result.uncommitted_output_bundles:
+    _unprocessed = True
```

Comment:
<!--thread_id:cc_131686170_t; commit:d61a79d86a74422660f4959a8abb348f9f1d5129; resolved:1-->
<!--section:context-quote-->
> **charlesccychen** wrote:
> We discussed this before--can you add a pipeline option to DirectOptions in options/pipeline_options.py to enable retrying of failed bundles?  This should be disabled by default because retrying of bundles containing state is not safe until state mutations are atomically committed per bundle (which we will do in a future change).

<!--section:body-->
Done.